[ENH] Change default of percentage error functions #1986

ciaran-g opened the pull request:
Fixes #1986. Introduced deprecation in percentage error functions for v.0.11.0 to precede a change of the default value for `symmetric` from `True` to `False` in v.0.12.0

<!--
Thanks for contributing a pull request! Please ensure you have taken a look
at our contribution guide: https://github.com/alan-turing-institute/sktime/blob/main/CONTRIBUTING.md
-->

#### Reference Issues/PRs
<!--
Example: Fixes #1234. See also #3456.

Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->

Fixes #1986 

#### What does this implement/fix? Explain your changes.
<!--
A clear and concise description of what you have implemented.
-->

Added depreciation notice for changing default in percentage error functions

#### Does your contribution introduce a new dependency? If yes, which one?

<!--
If your contribution does add a new hard dependency, we may suggest to initially develop your contribution in a separate companion package in https://github.com/sktime/ to keep external dependencies of the core sktime package to a minimum.
-->

No

#### What should a reviewer concentrate their feedback on?

<!-- This section is particularly useful if you have a pull request that is still in development. You can guide the reviews to focus on the parts that are ready for their comments. We suggest using bullets (indicated by * or -) and filled checkboxes [x] here -->

- correct use of the `@deprecated' function

#### Any other comments?
<!--
Please be aware that we are a loose team of volunteers so patience is necessary; assistance handling other issues is very welcome. We value all user contributions, no matter how minor they are. If we are slow to review, either the pull request needs some benchmarking, tinkering, convincing, etc. or more likely the reviewers are simply busy. In either case, we ask for your understanding during the review process.
-->

#### PR checklist
<!--
Please go through the checklist below. Please feel free to remove points if they are not applicable.
-->

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [ ] I've added unit tests and made sure they pass locally.

##### For new estimators
- [ ] I've added the estimator to the online documentation.
- [ ] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.


<!--
Thanks for contributing!
-->

fkiraly reviewed (changes requested):
@ciaran-g, there´s one linting failure coming from unsorted imports, otherwise looks good

fkiraly commented:
I'm not sure what is happening with all the tests here - some interaction with the deprecation decorator??

ciaran-g commented:
Hi @fkiraly - just checked the notebook tests. Hopefully the previous commit addressed the problem, but I'm not sure if it's the best way.

The tests were failing because there was an error thrown in the 01_forecasting notebook when calling this model tuning function
```
gscv = ForecastingGridSearchCV(forecaster=pipe, param_grid=param_grid, cv=cv, n_jobs=-1)
gscv.fit(y_train)
```
The deprecation decorator was causing an error with the default parallel backend in joblib. Basically it was failing to pickle the results. Changing the default parallel backend from `backend='loky'` to `backend='threading'` seems to do the trick for me anyway.

fkiraly commented:
@ciaran-g, how weird - thanks for tracking this down!

Regarding the change in the grid search, we should not just change the backend without warning - this would also be something that requires deprecation!

Instead, I would introduce a parameter for the parallelization back-end, if it does not exist, and change it in the notebook and/or test cases - but not in the default.

That way, we don't risk to break user code that uses instances of the grid search estimator.

ciaran-g commented:
Good shout Franz, just updated! Introduced the `backend` parameter, changed the call on the notebook, and the ForecastingGridSearchCV docstring example

fkiraly commented on sktime/forecasting/model_selection/_tune.py:
```diff
@@ -336,6 +342,9 @@ class ForecastingGridSearchCV(BaseGridSearch):
     return_train_score: bool, optional (default=False)
+    backend: str, optional (default=None)
+        Specify the parallelisation backend implementation in joblib, where
+        'loky' is used by default.
```
confusing - is it None or `"loky"`? Suggest: to avoid confusion, use the "effective" default which is `"loky"`.

fkiraly reviewed (changes requested):
with all the problems the decorators cause:

I'd suggest to move away from use of `deprecated` and instead raise a warning inside the function. Then everything should work again...

ciaran-g commented:
updated @fkiraly! left the `backend` argument open to the user, thought that would be useful anyway

fkiraly reviewed (changes requested):
I think this is good - would you mind reverting your changes to the forecaster tutorial?
As far as I can see, no content is changed or added.

fkiraly reviewed (approved):
I just did that if you don't mind, and will merge if/once tests pass.
